5-WebApp-AuthZ-Roles-Groups/Controllers/UserProfileController.cs

```diff
@@ -0,0 +1,38 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.Graph;
+using Microsoft.Identity.Web.Client;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using WebApp_OpenIDConnect_DotNet.Services.MicrosoftGraph;
+using Constants = WebApp_OpenIDConnect_DotNet.Infrastructure.Constants;
+
+namespace WebApp_OpenIDConnect_DotNet.Controllers
+{
+    public class UserProfileController : Controller
+    {
+        private readonly ITokenAcquisition tokenAcquisition;
+        private readonly IMSGraphService graphService;
+
+        public UserProfileController(ITokenAcquisition tokenAcquisition, IMSGraphService MSGraphService)
+        {
+            this.tokenAcquisition = tokenAcquisition;
+            this.graphService = MSGraphService;
+        }
+
+        [MsalUiRequiredExceptionFilter(Scopes = new[] { Constants.ScopeUserRead, Constants.ScopeDirectoryReadAll })]
```

Comment:
yes it does. it would have already been obtained during the initial sign-in 

---
In reply to: [272943999](https://github.com/Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2/pull/84#discussion_r272943999) [](ancestors = 272943999)